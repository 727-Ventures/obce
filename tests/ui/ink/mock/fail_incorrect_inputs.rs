#[obce::definition]
pub trait Trait {
    fn method(&self) -> u64;
}

#[obce::mock]
impl Trait for () {
    fn method(&self, val: u32) -> u64 {
        val as u64
    }
}

fn main() {}
