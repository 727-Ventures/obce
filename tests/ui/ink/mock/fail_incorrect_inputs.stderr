error[E0308]: mismatched types
 --> tests/ui/ink/mock/fail_incorrect_inputs.rs:6:1
  |
6 | #[obce::mock]
  | ^^^^^^^^^^^^^
  | |
  | expected `u32`, found `()`
  | arguments to this function are incorrect
  |
note: associated function defined here
 --> tests/ui/ink/mock/fail_incorrect_inputs.rs:8:8
  |
8 |     fn method(&self, val: u32) -> u64 {
  |        ^^^^^^
  = note: this error originates in the attribute macro `obce::mock` (in Nightly builds, run with -Z macro-backtrace for more info)
